packages/client-sdk-nodejs/src/internal/grpc/retry-interceptor.ts

```diff
@@ -107,7 +113,13 @@ export class RetryInterceptor {
                   next(status);
                   return;
                 }
-                nextDeadline = calculateDeadline(deadlineOffset);
+                // Do not exceed the overall deadline when setting the retry attempt's deadline.
+                nextDeadline = new Date(
+                  Math.min(
+                    overallDeadline.getTime(),
+                    calculateDeadline(deadlineOffset).getTime()
+                  )
+                );
```

Comment:
Would it make sense for this logic to be part of `calculateDeadline`? Eg an optional parameter with to clip the max value to.